### ISSUE TITLE
Transfer after not always working after comparison 

### ISSUE THREAD

**Issue by Vvkmnn** [PokemonGoF/PokemonGo-Bot#1418]
### Expected Behavior
Worse pokemon would be transferred after comparison.

### Actual Behavior

Worse pokemon are compared, then kept:
```
[10:38:10] A Wild Poliwag appeared! [CP 195] [Potential 0.53]
[10:38:10] IV [Stamina/Attack/Defense] = [5/4/15]
[10:38:12] Using Pokeball (chance: 47.34%)... (99 left!)
[10:38:13] Captured Poliwag! [CP 195] [Potential 0.53] [5/4/15]
[10:38:16] Owning better Poliwag [CP 441] [Potential 0.87] already!
[10:38:23] Spinning ...
[10:38:24] Loot:
[10:38:24] 50 xp
[10:38:24] - 3x Pokeball (Total: 102)
```

Doesn't always happen, some pokemon do get transferred:

```
[10:33:55] A Wild Weedle appeared! [CP 284] [Potential 0.47]
[10:33:55] IV [Stamina/Attack/Defense] = [3/13/5]
[10:33:58] Using Greatball (chance: 40.63%)... (48 left!)
[10:34:06] Captured Weedle! [CP 284] [Potential 0.47] [3/13/5]
[10:34:06] Owning weaker Weedle [CP 183] [Potential 0.51]. Replacing it with Weedle [CP 284] [Potential 0.47]!
[10:34:09] Weaker Weedle [CP 183] [Potential 0.51] has been exchanged for candy!
[10:34:14] [x] Found fort de8fae5f733349ab86ac2aa4a3a2251d.16 at distance 0.04km
[10:34:14] [x] Need to move closer to Pokestop
```

### Steps to Reproduce
Here's my current `config.json` 
```
    "release_pokemon": true,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or"},
      "// Example of always catching Rattata:": {},
      "// Rattata": { "always_catch" : true }
    },
    "release": {
      "any": {"keep_best_cp": true, "release_below_cp": 150, "release_below_iv": 0.8, "logic": "and"},

      "Snorlax": { "release_below_cp": 1000, "release_below_iv": 0.7, "logic": "and" },
      "Eevee": {"keep_best_iv" : true, "release_below_cp": 600, "logic": "and"},
      "Magikarp" : { "release_below_cp": 160, "keep_best_iv" : true, "logic": "and"},

      "Growlithe": { "release_below_cp": 600, "keep_best_iv": true, "logic": "and" },
      "Dragonair": { "release_below_cp": 800, "keep_best_iv": true, "logic": "and" },
      "Dragonair": { "release_below_cp": 600, "keep_best_iv": true, "logic": "and" },
      "Grimer": { "release_below_cp": 500, "keep_best_iv": true, "logic": "and" },

      "Bulbasaur": { "keep_best_iv": true },
      "Ivysaur": { "keep_best_iv": true },
      "Venusaur": { "keep_best_cp": true },

      "Charmander": { "keep_best_iv": true },
      "Charmeleon": { "keep_best_iv": true },
      "Charizard": { "keep_best_cp": true },

      "Squirtle": { "keep_best_iv": true },
      "Wartortle": { "keep_best_iv": true },
      "Blastoise": { "keep_best_cp": true },

      "// Example of always releasing Rattata:": {},
      "// Rattata": { "always_release" : true },
      "// Example of never releasing Snorlax:": {},
      "// Example of keeping stronger (based on CP) Pidgey:": {},
      "// Pidgey": { "keep_best_cp" : true },
      "// Example of keeping stronger (based on IV) Zubat:": {},
      "// Zubat": { "keep_best_iv" : true }
    }
```

### Other Information
OS: macOS 10.10.5
Git Commit: e25a7091f5f26eef17bc38f6f90df26ad199ffbc

### Fix
 #1418 and #1417 should be closed by #1423 or by #1395

**Comment by bigkraig** [comment]
It only transfers when it's replacing. When you already have a stronger pokemon it will keep the new one.

**Comment by bigkraig** [comment]
It looks like #1395 is going to fix it

**Comment by Vvkmnn** [comment]
Sweet. Thanks.

**Comment by Whosdaman** [comment]
Actually, I believe this is going to fix it #1423

**Comment by Vvkmnn** [comment]
Cheers. Updating.

**Comment by Vvkmnn** [comment]
Fixed with 522e35f541a82fcb0542f0be35aaf6dda28a7008

**Comment by Vvkmnn** [comment]
Close #1418

**Comment by Vvkmnn** [comment]
Closed #1418
